core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1951,11 +1950,8 @@ private[spark] class DAGScheduler(
               "Barrier stage will not retry stage due to testing config. Most recent failure " +
                 s"reason: $message"
             } else {
-              s"""$failedStage (${failedStage.name})
-                 |has failed the maximum allowable number of
-                 |times: $maxConsecutiveStageAttempts.
-                 |Most recent failure reason: $message
-               """.stripMargin.replaceAll("\n", " ")
+              s"$failedStage (${failedStage.name}) has failed the maximum allowable number of " +
+                s"times: $maxConsecutiveStageAttempts. Most recent failure reason: $message"
```

Comment:
Note that we don't append `\n` for  this "Most recent failure reason" because `message` already contains it:

```scala
val message = s"Stage failed because barrier task $task finished unsuccessfully.\n" +
  failure.toErrorString
```